### PR TITLE
fix: fixing tear-off positions (ARTP-628)

### PR DESCRIPTION
**Pull request by sizov**
- Tiles which were torn off by dragging will now appear at the position where user dragged them
- Tiles which were torn off by clicking on "pop out" button will appear at the position of the click

### CODE REVIEW

File: src/client/src/rt-components/tear-off/TearOff.tsx [hunk 1]
```diff
@@ -82,10 +100,9 @@ const TearOff: React.FC<TearOffProps> = ({ render, externalWindowProps, tornOff,
     return (
       <DragWrapper
         draggable={allowTearOff}
-        onDragEnd={(event: React.DragEvent<HTMLDivElement>) => {
-          popOut(event.screenX, event.screenY)
-        }}
-        onDragStart={onDragStart}
+        onDragEnd={(event: React.DragEvent<HTMLDivElement>) => popOut(event.screenX, event.screenY)}
+        onDragStart={createDragImage}
+        onMouseDownCapture={onMouseDownCapture}
```
Comment (sizov) [inline src/client/src/rt-components/tear-off/TearOff.tsx]: this is a new event where we will save coordinates of mouse relative to the tile. These coordinates will be used when we tearing things off

File: src/client/src/rt-components/tear-off/TearOff.tsx [hunk 2]
```diff
@@ -57,23 +57,41 @@ export interface TearOffProps {
...
-      dispatch(LayoutActions.updateContainerVisibilityAction({ name: windowName, display: true })),
+      dispatch(LayoutActions.updateContainerVisibilityAction({name: windowName, display: true})),
     [windowName, dispatch],
   )
 
+  const onMouseDownCapture = (event: React.MouseEvent<HTMLDivElement>) => {
+    const eventTarget = event.currentTarget as HTMLDivElement
+
+    // calculating mouse position relative to the torn off widget
+    const clientRect = eventTarget.getBoundingClientRect()
```
Comment (sizov) [inline src/client/src/rt-components/tear-off/TearOff.tsx]: saving mouse coordinate relative to the tile

File: src/client/src/rt-components/tear-off/TearOff.tsx [hunk 3]
```diff
@@ -57,23 +57,41 @@ export interface TearOffProps {
...
+  const {allowTearOff} = usePlatform()
+  const targetMouseXRef = useRef<number>()
+  const targetMouseYRef = useRef<number>()
+
   const dispatch = useDispatch()
   const windowName = externalWindowProps.config && externalWindowProps.config.name
   const popOut = useCallback(
-    (x?: number, y?: number) =>
+    (mouseScreenX?: number, mouseScreenY?: number) => {
+      const popOutX = typeof targetMouseXRef.current !== 'undefined' && typeof mouseScreenX !== 'undefined' ?
```
Comment (sizov) [inline src/client/src/rt-components/tear-off/TearOff.tsx]: using coordinates saves previously in `mouseDownCapture` handler

File: src/client/src/rt-platforms/openFin/adapter/window.ts [hunk 4]
```diff
@@ -74,32 +74,42 @@ export function createPlatformWindow(getWindow: () => Promise<fin.OpenFinWindow>
...
+type OpenfinWindowPosition = Pick<fin.WindowOption, 'defaultLeft' | 'defaultTop'>;
 
-  const childWindows = await getChildWindows()
-  const hasChildWindows = childWindows && childWindows.length > 0
-  const shouldBeDefaultCentered = !hasChildWindows && (!config.x && !config.y)
-
-  let updatedPosition = {
-    defaultLeft: config.x ? config.x : undefined,
-    defaultTop: config.y ? config.y : undefined,
+async function getOpenfinWindowPosition(config: DesktopWindowProps, childWindows?: fin.OpenFinWindow[]): Promise<OpenfinWindowPosition> {
```
Comment (sizov) [inline src/client/src/rt-platforms/openFin/adapter/window.ts]: moving calculating position to a new function. Here the change is to NOT do cascading windows when there are explicit coordinates from the function call

File: src/client/src/rt-platforms/openFin/adapter/window.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable no-undef */
-import { WindowConfig } from '../../types'
```
Comment (rikoe) [inline src/client/src/rt-platforms/openFin/adapter/window.ts]: Do we have listing settings that control spacing around braces (since it is changing)?

Comment (sizov) [reply src/client/src/rt-platforms/openFin/adapter/window.ts]: it's my intelij config. will changem thanks for pointing out

**Pull request merged**